### ISSUE TITLE
Improve GUI images with low resolution

### ISSUE THREAD

**Issue by hhyyrylainen** [Revolutionary-Games/Thrive#1581]
The following things in the GUI are too low resolution at 4K and look blurry:
- [ ] Main menu background thrive logo
- [ ] checkbox selected and deselected images
- [ ] spinbox up/down arrows
- [ ] slider circular part
- [ ] environment and compound panels the compact and normal view icons
- [ ] in the microbe stage all the bottom left row icons
- [ ] reproduction indicator bars, the circle, and the central dna icon
- [ ] close x in dialogs
- [ ] the trashcan icon / button
- [ ] all of the patch map, patch icons
- [ ] patch condition up and down arrows
- [ ] colour picker HSV and RAW off / on buttons
- [ ] symmetry icons
- [ ] undo and redo in editor
- [ ] the plus icon in new cell button

Note: there might be places where changing the icon resolution breaks things so there's a chance that godot GUI scenes need tweaking as well as just the image files.

**Comment by skyace65** [comment]
Any of these images that don't have an SVG file are going to have to be traced. There was someone on discord who might have some of these.

**Comment by hhyyrylainen** [comment]
It was discussed on our dev discord that apparently godot does support svg import, but what it does it just automatically makes a png out of it, hopefully the resolution of which is configurable.

Related to the icons, I have a 1000x1000 thrive logo image which might be a replacement for the menu background logo:
![Thrivelogobig](https://user-images.githubusercontent.com/3796411/92312579-bd80f980-efca-11ea-9533-2a172c92f1ad.png)
(it's white on transparent so it isn't visible on github)

**Comment by Kasterisk** [comment]
The slider grabber icon seems to have to be a small res for some reason, there isn't a setting to control its size (at least from what I've seen) on the theme styling options, so if it were to be made big it will take up a lot of space:
![2020-09-06_21 23 57 2280](https://user-images.githubusercontent.com/54026083/92327949-86245280-f087-11ea-8bb3-66f8a89b8987.png)
The one above was the original size DonGororo made, but I had to reduce it to 15x15px to make it fit nicely.

**Comment by hhyyrylainen** [comment]
That seems like something Godot should fix. Could you see if you can find an open bug about this, and maybe open a bug / godot proposal for allowing to set the size for the slider?

**Comment by Kasterisk** [comment]
Looks like there is already a godot proposal open for this https://github.com/godotengine/godot-proposals/issues/660

**Comment by skyace65** [comment]
It might be worth trying what Calinou suggested in that thread

> Set the project width and height to the desired resolution in the Project Settings, then set Test Width and Test Height to the default window size you'd like.

**Comment by hhyyrylainen** [comment]
Would that work correctly an exported game as well? I think I had a look at the test width and based on the tooltip for it didn't consider it a suitable choice.
Would it also still be easy to edit the GUI so that it works fine at 720p?
I think we might need to set the project resolution to at least 1440p to not have visible blurriness at 4k. So it might be then difficult to actually make sure that the game is playable at 720p (as that's the minimum resolution I want thrive to support).

**Comment by skyace65** [comment]
Theoretically it should. I'll test it out when I get the chance.

**Comment by Kasterisk** [comment]
@skyace65 Narotiza shared some svg files for the new editor and current microbe stage UI on the dev forum (which I use to get some new icons for the editor UI rework) https://drive.google.com/drive/folders/19B-ji46CMayUiXWGFdy7UM7Bl9AC70vZ?usp=sharing And here's the svg for the checkboxes and grabbers [UI_Buttons.zip](https://github.com/Revolutionary-Games/Thrive/files/5181434/UI_Buttons.zip)

Also, the reproduction indicator bar, undo, redo and the plus icon are probably gonna be obsolete as it has been replaced in my branch with the newer ones.

**Comment by skyace65** [comment]
Looks like this should be easier to do now with this PR merged https://github.com/godotengine/godot/pull/52137. I'll look into it this weekend but I'm making this comment in case I don't get around to it

**Comment by hhyyrylainen** [comment]
If Godot has now fixed this, that is excellent news. We plan to update to the latest Godot version next week or we might wait for 3.5 to be released.

**Comment by skyace65** [comment]
The fix is actually in 3.4 so thrive can use it right now. I'm currently testing it.

**Comment by skyace65** [comment]
Unfortunately this solution isn't going to work for thrive. For stuff like the main menu logo the scale property gets completely disregarded because it's in a center container. The slider grabber has no scale property so we're stuck with the "it gets too big" issue.

I think we're just going to have to wait for proper SVG support. That is something wanted by the community but it's not being actively worked on as far as I know.

**Comment by Kertit** [comment]
'Been a while no ones cared about those images. Are they still a thing to care about?

**Comment by hhyyrylainen** [comment]
I still care that parts of the GUI are pixelated. And anyone with a high resolution monitor will be able to notice the weird different clarity of GUI elements on the same screen.

**Comment by Kertit** [comment]
That Is true as I do with my 1440p monitor. 
Going to see what I can do

**Comment by Kertit** [comment]
erw, yeah I don't how its done in godot but higher res element provoke this 
![image](https://github.com/Revolutionary-Games/Thrive/assets/35419629/30adb07d-2943-4cd5-b028-5d0798a4dc1e)

**Comment by hhyyrylainen** [comment]
Yes, that's the original reason why things are low-res. Referring back to those Godot proposal and change linked above, it should now be possible to specify in the style the "expand" option for the texture, which contrary to the name also allows an image to be scaled down from its native resolution.

If you can't find how to get it working some programmer may be needed to look into this.

**Comment by Kertit** [comment]
well i've tried to use that expand option but the icon keeps diappearing when I change value (so it works well)>save>quit>reopen. 
Will pack those icon in DevCenter someone else will probably don't get that problem

**Comment by Kertit** [comment]
Seems like some of the listed elements are good already. The list might not be up to date?

**Comment by hhyyrylainen** [comment]
Which ones are already good? I'm asking so that I can double check those ones.

A few comments on the icons you already posted:

- The symmetry icons have a bunch of whitespace in them. The new icons should preserve the existing ratio of whitespace to the icon at the center. Also the new icon files are actually smaller (100x100) than the ones they would replace (240x240). So I think those need another iteration of making those again at maybe 800x800 resolution (which will give the center image more resolution to work with even when preserving the whitespace ratio).

- For the other icons (check), I think doing them at 240x240 resolution (or 200x200) would be the best as that seems to be used for other buttons (and various other icons).

**Comment by Kertit** [comment]
List of already good thing (to me):

the trashcan icon / button
-"colour picker HSV and RAW off / on buttons"  (I don't know where that button is)
-undo and redo in editor
-"the plus icon in new cell button"  (I don't know where that button is)

- As for the symmetry Icons, they are 80x80 not 240. The only 240px icon is the 1xsymmetry. on my screen (1440p) they looks good. But could be bigger res if you really want to. that's no problem I've created them in 2000px then just resized them.
Will check that whitespace thing I'm not sure what you mean.

- Will double the size then.

**Comment by hhyyrylainen** [comment]
> the trashcan icon / button

agree

>  -"colour picker HSV and RAW off / on buttons" (I don't know where that button is)

this uses the inbuilt Godot texture, which is very low res. We should do our own. This is currently hidden in the game as the microbe editor colour picker has these turned off.

>  -undo and redo in editor

agree

>  -"the plus icon in new cell button" (I don't know where that button is)

I think it is "NewIcon.png" which is 40x40

> May I ask you to rename the element list with the actual file name? I can't find some of them/not sure what is what

I'll see if I can find them and update the list.

**Comment by HexapodPhilosopher** [comment]
I did a new SVG version of NewIcon.png, but can't find where it's used. Is it defunct now?

**Comment by hhyyrylainen** [comment]
It doesn't seem to be used anywhere; replacing it with any kind of svg replacement should be fine. I wouldn't delete it outright as it might be useful as an indicator on top of another icon (in a corner) or something like that.

**Comment by hhyyrylainen** [comment]
Referring to the revert to blank cell button (which the new button description here talks about), that's actually now using the icon `res://assets/textures/gui/bevel/clockButton.png` (and its variants).

**Comment by hhyyrylainen** [comment]
Hopefully I marked off the right items from the list after #5875 

It's starting to look like there's not much left on this issue.
